core/src/test/java/org/kohsuke/stapler/ClassDescriptorTest.java

```diff
@@ -28,6 +28,32 @@
         assertEquals("[a, b, x]",Arrays.asList(names).toString());
     }
 
+    @Test public void loadParameterNamesFromReflection() throws Exception {
+        // collect test cases
+        Map<String,Method> testCases = new HashMap<String,Method>();
+        for (Method m : ClassDescriptorTest.class.getDeclaredMethods())
+            if (m.getName().startsWith("methodWith"))
+                testCases.put(m.getName().substring(10), m);
+        // expected results
+        Map<String,String[]> expected = new HashMap<String,String[]>();
+        expected.put("NoParams", new String[0]);
+        expected.put("NoParams_static", new String[0]);
+        expected.put("ManyParams", new String[] { "a", "b", "c", "d", "e", "f", "g", "h", "i" });
+        expected.put("Params_static", new String[] { "abc", "def", "ghi" });
+        // run tests
+        for (Map.Entry<String,String[]> entry : expected.entrySet()) {
+            Method testMethod = testCases.get(entry.getKey());
+            assertNotNull("Method missing for " + entry.getKey(), testMethod);
+            String[] result = ClassDescriptor.loadParameterNamesFromReflection(testMethod);
+            assertNotNull("Null result for " + entry.getKey(), result);
+            if (!Arrays.equals(entry.getValue(), result)) {
```

Comment:
Could use a standard `Matcher` for this I think.

Reply:
I was just trying to change the existing code as little as possible. This is pretty much a copy of `loadParametersFromAsm`. There is a lot that can be cleaned up in this repository. Once the dependencies are up-to-date I'll start looking into cleanups like this.

Reply:
In particular the use of reflection in the ASM test is unnecessary (fixed in #226).